kcoin/genesis/genesis_generator_test.go

```diff
@@ -0,0 +1,300 @@
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.MaxNumValidators = ""
+				return command
+			},
+			ExpectedError: ErrEmptyMaxNumValidators,
+		},
+		{
+			TestName: "Empty unbonding period of days",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.UnbondingPeriod = ""
+				return command
+			},
+			ExpectedError: ErrEmptyUnbondingPeriod,
+		},
+		{
+			TestName: "Empty wallet address of genesis validator",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.WalletAddressGenesisValidator = ""
+				return command
+			},
+			ExpectedError: ErrEmptyWalletAddressValidator,
+		},
+		{
+			TestName: "Invalid wallet address less than 20 bytes with Hex prefix",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.WalletAddressGenesisValidator = "0xe2ac86cbae1bbbb47d157516d334e70859a1be"
+				return command
+			},
+			ExpectedError: ErrInvalidWalletAddressValidator,
+		},
+		{
+			TestName: "Invalid wallet address less than 20 bytes without Hex prefix",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.WalletAddressGenesisValidator = "e2ac86cbae1bbbb47d157516d334e70859a1be"
+				return command
+			},
+			ExpectedError: ErrInvalidWalletAddressValidator,
+		},
+		{
+			TestName: "Empty prefunded accounts",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.PrefundedAccounts = []PrefundedAccount{}
+				return command
+			},
+			ExpectedError: ErrEmptyPrefundedAccounts,
+		},
+		{
+			TestName: "Prefunded accounts does not include validator address",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.PrefundedAccounts = []PrefundedAccount{
+					{
+						WalletAddress: "0xaaaaaacbae1bbbb47d157516d334e70859a1bee4",
+						Balance:       "15",
+					},
+				}
+				return command
+			},
+			ExpectedError: ErrWalletAddressValidatorNotInPrefundedAccounts,
+		},
+		{
+			TestName: "Prefunded accounts has invalid account.",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.PrefundedAccounts = []PrefundedAccount{
+					{
+						WalletAddress: "0xe2ac86cbae1bbbb47d157516d334e70859a1bee4",
+						Balance:       "15",
+					},
+					{
+						WalletAddress: "0xe286cbae1bbbb47d157516d334e70859a1bee4",
+						Balance:       "15",
+					},
+				}
+				return command
+			},
+			ExpectedError: ErrInvalidAddressInPrefundedAccounts,
+		},
+		{
+			TestName: "Invalid consensus engine.",
+			InvalidCommandFromValid: func(command GenesisOptions) GenesisOptions {
+				command.ConsensusEngine = "fakeConsensus"
+				return command
+			},
+			ExpectedError: ErrInvalidConsensusEngine,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.TestName, func(t *testing.T) {
+			_, err := GenerateGenesis(test.InvalidCommandFromValid(baseValidCommand))
+			if err != test.ExpectedError {
```

Comment:
`assert.EqualError(t, test.ExceptedError, err)`

I think this will log the case under test